src/vos/evtree.c

```diff
@@ -3215,6 +3213,85 @@ int evt_delete(daos_handle_t toh, const struct evt_rect *rect,
 	/* No need for evt_ent_array_fill as there will be no allocations
 	 * with 1 entry in the list
 	 */
+	return in_tx ? rc : evt_tx_end(tcx, rc);
+}
+
+int evt_delete(daos_handle_t toh, const struct evt_rect *rect,
+	       struct evt_entry *ent)
+{
+	struct evt_context	*tcx;
+
+	tcx = evt_hdl2tcx(toh);
+	if (tcx == NULL)
+		return -DER_NO_HDL;
+
+	return evt_delete_internal(tcx, rect, ent, false);
+}
+
+int
+evt_remove_all(daos_handle_t toh, const struct evt_extent *ext,
+	       daos_epoch_t epoch)
```

Comment:
Would it more flexible to pass in an epoch range?

Reply:
You are right.  We need a new vos API to pass one.   The issue is if we are doing EC aggregation on a snapshot, we don't want to affect the partial stripes written before the snapshot range.   I'll need a new vos api afterall :-)